shopfloor/services/location_content_transfer.py

```diff
@@ -764,9 +764,10 @@ def set_destination_line(
         self._lock_lines(move_line)
 
         move_line.qty_done = quantity
-        remaining_move_line = move_line._split_partial_quantity()
+        # If a move line reference a move linked to others move lines (different
+        # locations), we have to split the move to process only the lines related to
+        # the current location.
         move_line._extract_in_split_order({"user_id": self.env.uid})
```

Comment:
How is that possible as when you scan the location at the start, you isolate the move lines from that location in a split order? So at this stage inside the picking we should only have move lines from that location.

In set_destination_all and set_destination_package, I think such is not done. We should have the same behavior in all 3 cases.

I would have dropped that line also unless you can reproduce the use case.

Reply:
It's required by a test (https://github.com/OCA/wms/pull/679/files#diff-d468159958b3bfcb32afaa09c6a77a69e1811aa92d2abe6d4022f1be66a9ad19R637) but reading the code, It seems that the test is not right since the move_line to process is not selected by using the `scan_location` method and therefore the initial split is not done... I'll fix the test and remove the line.

Reply:
@jbaudoux It's fixed into the last commit https://github.com/OCA/wms/pull/679/commits/dbab97dca13c8402b45275840363f8b54ec7ce06